enterprise/views.py

```diff
@@ -499,38 +505,65 @@ def get_enterprise_course_enrollment_page(self, request, enterprise_uuid, course
             logger.error('Unable to find course details for course ID: %s', course_id)
             raise Http404
 
-        platform_name = configuration_helpers.get_value('PLATFORM_NAME', settings.PLATFORM_NAME)
+        try:
+            enrollment_client = EnrollmentApiClient()
+            modes = enrollment_client.get_course_modes(course_id)
+        except HttpClientError:
+            logger.error('Failed to determine available course modes for course ID: %s', course_id)
+            raise Http404
+
+        # This is protected by the enterprise_login_required decorator; we don't need to guard it here
+        enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_uuid)  # pylint: disable=no-member
+
+        course_modes = []
+
+        audit_modes = getattr(settings, 'ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES', ['audit'])
+
+        for mode in modes:
+            if mode['min_price']:
+                price_text = '${}'.format(mode['min_price'])
+            else:
+                price_text = self.context_data['free_price_text']
+            if mode['slug'] in audit_modes:
+                description = self.context_data['audit_text']
+            else:
+                description = self.context_data['verified_text']
+            course_modes.append({
+                'mode': mode['slug'],
+                'title': mode['name'],
+                'original_price': price_text,
+                'final_price': price_text,
+                'description': description,
+                'premium': mode['slug'] not in audit_modes
```

Comment:
Please add `sku` too which will be required for directing the learner to product basket page or to get enterprise coupons w.r.t the `sku` product.